loader/extends.go

```diff
@@ -77,15 +77,15 @@ func applyServiceExtends(ctx context.Context, name string, services map[string]a
 
 	var base any
 	if file != nil {
-		filename = file.(string)
-		services, err = getExtendsBaseFromFile(ctx, ref, filename, opts, tracker)
+		refFilename := file.(string)
```

Comment:
I wonder about the impact not assigning `filename` anymore: 
`tracker.Add(filename, name)` will then fail to capture the target file

Reply:
assigned the name to `filename` after the func call